OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs

```diff
@@ -178,6 +181,96 @@ public async Task<ProviderDto> Update(ProviderDto providerDto, string userId)
         public async Task<Guid> GetProviderIdForWorkshopById(Guid workshopId) =>
             await workshopServiceCombiner.GetWorkshopProviderId(workshopId).ConfigureAwait(false);
 
+        public async Task<ProviderStatusDto> UpdateStatus(ProviderStatusDto dto, string userId)
+        {
+            _ = dto ?? throw new ArgumentNullException(nameof(dto));
+
+            var provider = await providerRepository.GetById(dto.ProviderId).ConfigureAwait(false);
+
+            if (provider is null)
+            {
+                logger.LogError($"Provider(id) {dto.ProviderId} not found. User(id): {userId}");
```

Comment:
we do like to use error logs throughout the application :) but here probably nothing bad has happened, it might be just an user error or someone deleted the provider before admin could click approve. As user experience or application functioning is not impacted - maybe use `info` or `warning`?